base/expr.jl

```diff
@@ -407,7 +407,7 @@ function constprop_setting(@nospecialize setting)
     elseif setting === :none
         return :no_constprop
     end
-    throw(ArgumentError(LazyString("@constprop "), setting, "not supported"))
+    throw(ArgumentError(LazyString("`@constprop ", setting, "` not supported")))
```

Comment:
Refrain from `LazyString`, as it seems unnecessary. This improves consistency with the error handling for `@assume_effects` below.

```suggestion
    throw(ArgumentError("`@constprop $setting` not supported"))
```

Reply:
Could you elaborate on the issue here that is caused by `LazyString`?

Reply:
`LazyString` doesn't cause any issue. I replaced it with string interpolation because:
* interpolation is simpler for anyone reading the code, and using `LazyString` seems unnecessary, considering that this is a macro so the usual performance reasons for using it don't apply as far as I understand
* that's what's done below for `@assume_effects`: https://github.com/JuliaLang/julia/blob/eee709e8330a508d11729308dc3605fee89db143/base/expr.jl#L783-L791

Reply:
Thanks, that makes sense